src/offset/local/windows.rs

```diff
@@ -72,41 +78,53 @@ impl LocalSysTime {
         // is initialized.
         let st = unsafe { now.assume_init() };
 
-        Self::from_local_time(st).expect("Current local time must exist")
+        Self::from_local_time(st, false).expect("Current local time must exist")
     }
 
-    fn from_utc_time(utc_time: SYSTEMTIME) -> Result<Self, Error> {
+    fn from_utc_time(utc_time: SYSTEMTIME, shifted: bool) -> Result<Self, Error> {
         let local_time = utc_to_local_time(&utc_time)?;
         let utc_secs = system_time_as_unix_seconds(&utc_time)?;
         let local_secs = system_time_as_unix_seconds(&local_time)?;
         let offset = (local_secs - utc_secs) as i32;
-        Ok(Self { inner: local_time, offset })
+        Ok(Self { inner: local_time, offset, shifted })
     }
 
-    fn from_local_time(local_time: SYSTEMTIME) -> Result<Self, Error> {
+    fn from_local_time(local_time: SYSTEMTIME, shifted: bool) -> Result<Self, Error> {
         let utc_time = local_to_utc_time(&local_time)?;
         let utc_secs = system_time_as_unix_seconds(&utc_time)?;
         let local_secs = system_time_as_unix_seconds(&local_time)?;
         let offset = (local_secs - utc_secs) as i32;
-        Ok(Self { inner: local_time, offset })
+        Ok(Self { inner: local_time, offset, shifted })
     }
 
     fn datetime(self) -> DateTime<Local> {
         let st = self.inner;
 
-        let date =
-            NaiveDate::from_ymd_opt(st.wYear as i32, st.wMonth as u32, st.wDay as u32).unwrap();
-        let time = NaiveTime::from_hms(st.wHour as u32, st.wMinute as u32, st.wSecond as u32);
+        let year = if self.shifted { st.wYear - 1600 } else { st.wYear };
+
+        let date = NaiveDate::from_ymd_opt(year as i32, st.wMonth as u32, st.wDay as u32).unwrap();
+        let time = NaiveTime::from_hms_milli_opt(
+            st.wHour as u32,
+            st.wMinute as u32,
+            st.wSecond as u32,
+            st.wMilliseconds as u32,
+        )
+        .unwrap();
 
         let offset = FixedOffset::east_opt(self.offset).unwrap();
         DateTime::from_utc(date.and_time(time) - offset, offset)
     }
 }
 
-fn system_time_from_naive_date_time(dt: &NaiveDateTime) -> SYSTEMTIME {
-    SYSTEMTIME {
+fn system_time_from_naive_date_time(dt: &NaiveDateTime) -> (SYSTEMTIME, bool) {
```

Comment:
Is it possible to add test cases for only this function `system_time_from_naive_date_time`?
I can see this function is indirectly tested elsewhere but it seems good to me have tests focused solely on this function.